catalogs/catalog-jdbc-mysql/src/test/java/com/datastrato/gravitino/catalog/mysql/integration/test/CatalogMysqlIT.java

```diff
@@ -114,12 +116,14 @@ public void startup() throws IOException {
       JdbcDriverDownloader.downloadJdbcDriver(DOWNLOAD_JDBC_DRIVER_URL, tmpPath.toString());
     }
 
-    MYSQL_CONTAINER =
-        new MySQLContainer<>(mysqlImageName)
-            .withDatabaseName(TEST_DB_NAME)
-            .withUsername("root")
-            .withPassword("root");
-    MYSQL_CONTAINER.start();
+    if (mysqlImageName.equals("mysql:5.7")) {
+      containerSuite.startMySQLVersion5Container(CatalogMysqlIT.class);
+      MYSQL_CONTAINER = containerSuite.getMySQLVersion5Container();
```

Comment:
Have you forgotten to set the database to `MySQLContainer`, please see:

```
 MYSQL_CONTAINER =
        new MySQLContainer<>(mysqlImageName)
            .withDatabaseName(TEST_DB_NAME)
            .withUsername("root")
            .withPassword("root");
```

If we do not set `.withDatabaseName(TEST_DB_NAME)` then, https://github.com/datastrato/gravitino/pull/2813/files#diff-bbb55642e9d5ac96f87540a0418d79b8bfb16cb747d8f0a74bfb56c2a1a98a46R50:~:text=String%20jdbcUrl%20%3D%20mySQLContainer.getJdbcUrl()%3B will get an URL that do not contain database information, that is why the IT fails.

Reply:
I'll try to fix that. Thanks : )

Reply:
But even if this is the cause, why my Intellij works but cli didn't ?

Reply:
But even if this is the cause, why my Intellij works but cli didn't ?

Reply:
@unknowntpo 
Have you set the parameter '-PjdbcBackend' in Intellij? If not, Gravitino will not use MySQL as the storage backend.

Reply:
you're right, after setting `-PjdbcBackend` in run configuration, I got the same result as running with cli.
![image](https://github.com/datastrato/gravitino/assets/10594519/9fffb75c-1d5a-4bcf-b140-182aea328614)

Reply:
I have a small question:
Is it a good idea that we maintain a `.idea` in our repo ?
Because for developers that are not very familiar with the code space of integration test, it's very likely that they don't event know parameters like `-PjdbcBackend` exists.

Reply:
Let me think about it.

Reply:
It's unnecessary to maintain `.idea`, we can use `@ParameterizedTest` annotation to achieve this

Reply:
It seems like we have these four combinations:

- embedded, jdbcBackend
- embedded, kvBackend
- deploy, jdbcBackend
- deploy, kvBackend

And that's why we need to run Backend Integration Test for 12 times (with 3 java version).

So I think we can do it this way:

```
@ParameterizedTest
    @CsvSource({
        "embedded, jdbcBackend",
        "embedded, kvBackend",
        "deploy, jdbcBackend",
        "deploy, kvBackend"
    })
    void startIntegrationTest(String testMode, String backend) {
```

But I found that `@BeforeAll` seems to be executed only one time.

So my question is: Is it a way to parameterize beforeAll ?

After I googled it and found this comments, but I don't know it works or not.

https://github.com/junit-team/junit5/issues/3157#issuecomment-1435009129

Reply:
@unknowntpo 
Can you create a new issue and discuss it in that forum?